pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOrderByOperator.java

```diff
@@ -117,6 +119,30 @@ public String toExplainString() {
     return stringBuilder.append(')').toString();
   }
 
+  private void pruneOrderByColumns() {
+    // Order by col1, col2 limit k will take all the n matching rows
+    // and add it to a priority queue of size k.
+    // This is nlogk operation which can be quite expensive for a large n.
+    // This function, prunes out any order by clause that matches columns sorting order.
+    // In the above example, if the docs are already sorted by col1, col2 then there is no need for any
+    // sorting at all (only limit is needed). When only col1 is sorted, then we need to use only col2 for ordering.
```

Comment:
I don't follow the logic here. Say col1 is sorted, col2 is not. If we apply this logic, will the final result only order on col2?

Reply:
Actually, the comment is wrong. We apply the optimization (early termination) only if all columns are sorted. Lemme fix it up